jwala-services/src/main/java/com/cerner/jwala/service/custom/logging/WinSvcPasswordMaskingLayout.java

```diff
@@ -0,0 +1,84 @@
+package com.cerner.jwala.service.custom.logging;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.spi.LoggingEvent;
+
+import java.util.*;
+
+/**
+ * Masks the Windows Service Password (svc_password)
+ * Created by Jedd Cuison on 6/6/2017
+ */
+public class WinSvcPasswordMaskingLayout extends PatternLayout {
+
+    private String forClass;
+    private Set<String> forMethods; // Search is faster for Set
+
+    @Override
+    public String format(final LoggingEvent event) {
+
+        // Since this class specifically handles windows service passwords (svc_password) found in the logs,
+        // we filter out events, classes and methods that we know won't log svc_password to minimize the impact
+        // of this layout on the application's performance
+        if (event.getLevel() != Level.DEBUG || isClassExcluded(event) || isMethodExcluded(event)) {
+            return super.format(event);
+        }
+
+        final String msg = event.getRenderedMessage();
+        String maskedMsg = StringUtils.EMPTY;
+
+        // This layout is specific for svc_password masking therefore the code is straight forward and not generic
+        if (StringUtils.indexOf(msg, "install-service.sh") > 0) {
+            int index = msg.lastIndexOf(' ', msg.lastIndexOf(' ') - 1);
+            maskedMsg = msg.substring(0, index) + " ******** \"";
+        } else if (StringUtils.contains(msg, "set svc_password=")) {
```

Comment:
Can you move the hard coded strings as constants, so we can change them easyling if required?